tests/tel2puml/otel_to_pv/data_holders/sql_data_holder/test_sql_dataholder.py

```diff
@@ -703,3 +695,201 @@ def test_find_unique_graphs(
+
+    # Compare the expected and streamed data
+    assert len(expected_events) == len(streamed_events)
+
+    for event_id, expected_event in expected_events.items():
+        assert event_id in streamed_events
+        assert expected_event == streamed_events[event_id]
+
+    # Test 2: Stream from empty database
+    empty_sql_data_holder = SQLDataHolder(
+        config=SQLDataHolderConfig(
+            db_uri="sqlite:///:memory:", batch_size=10, time_buffer=30
+        )
+    )
+
+    with empty_sql_data_holder:
+        result = empty_sql_data_holder.stream_data()
+
+        all_events = []
+        for job_name, job_id_gen in result:
+            for otel_event_gen in job_id_gen:
+                for otel_event in otel_event_gen:
+                    all_events.append(otel_event)
+
+        assert len(all_events) == 0
+
+    # Test 3: Stream from large dataset
+    large_sql_data_holder = SQLDataHolder(
+        config=SQLDataHolderConfig(
+            db_uri="sqlite:///:memory:", batch_size=1000, time_buffer=30
+        )
+    )
+
+    num_jobs = 10
+    events_per_job = 10000  # Total of 100,000 events
+
+    with large_sql_data_holder:
+        for job_index in range(num_jobs):
+            job_name = f"job_{job_index}"
+            job_id = f"job_id_{job_index}"
+            for event_index in range(events_per_job):
+                otel_event = OTelEvent(
+                    job_name=job_name,
+                    job_id=job_id,
+                    event_type=f"event_type_{event_index}",
+                    event_id=f"event_{job_index}_{event_index}",
+                    start_timestamp=1723544132228288000 + event_index,
+                    end_timestamp=1723544132228288000 + event_index + 10,
+                    application_name="test_application",
+                    parent_event_id=None,
+                    child_event_ids=[],
+                )
+                large_sql_data_holder.save_data(otel_event)
+
+    # Stream data and verify counts
+    total_events_streamed = 0
+    with large_sql_data_holder:
+        result = large_sql_data_holder.stream_data()
+        for job_name, job_id_gen in result:
+            for otel_event_gen in job_id_gen:
+                event_count = 0
+                for otel_event in otel_event_gen:
+                    event_count += 1
+                assert event_count == events_per_job
+                total_events_streamed += event_count
+
+    assert total_events_streamed == num_jobs * events_per_job
+
+
+def test_stream_job_name_batches(
+    sql_data_holder_with_multiple_otel_job_names: SQLDataHolder,
+) -> None:
+    """Test stream_job_name_batches method."""
+    # Test 1: Stream all data
+    sql_data_holder = sql_data_holder_with_multiple_otel_job_names
+    with sql_data_holder.session as session:
+        otel_event_gen = sql_data_holder.stream_job_name_batches(session)
+        all_events = list(otel_event_gen)
+
+    assert len(all_events) == 20
```

Comment:
just for this one can we make sure we have the sequence of job names and job ids correct as we should be checking that its all correctly ordered in the output i.e.

```
assert all(
    event.job_name == expected_job_name and event.job_id == expected_job_id
    for event, expected_job_name, expected_job_id in zip(
        all_events,
        ["test_name"] * 10 + ["test_name_1"] *10),
        [f"test_id_{i}" for i in range(10) for _ in range(2)]
)
```

Reply:
Change made